srv/salt/_modules/cephdisks.py

```diff
@@ -2,82 +2,429 @@
 
 import os
 import re
-import pprint
+import xml.etree.ElementTree as et
 from glob import glob
 from subprocess import Popen, PIPE
+import logging
 
+log = logging.getLogger(__name__)
 
-def list():
+
+class HardwareDetections(object):
     """
-    Find all unpartitioned and allocated osds.  Return hwinfo dict.
+    Custom Salt module to detect suitable disks for the ceph deployment
     """
-    drives = []
-    for path in glob('/sys/block/*/device'):
-        base = os.path.dirname(path)
-        device = os.path.basename(base)
-
-        # Skip partitioned, non-osd drives
-        partitions = glob(base + "/" + device + "*")
-        if partitions:
-            for p in partitions:
-                ids = [ re.sub('\D+', '', p) for p in partitions ]
-            if not _osd("/dev/" + device, ids):
-                continue
 
-        # Skip removable media
+    def __init__(self, **kwargs):
+        """
+        kwargs adds the possibility to pass custom raidcontroller
+        names to the class.
+        args:
+            hw_raid(bool): Manually set hw_raid True if this class can't detect it
+            hw_raid_name(str): Manually set the hw_raid_ctrls name
+            software_raid(bool): Manually set if you have sw raid and the class
+                                            fails to detect it.
+        """
+        self.overwrite_detection_method = kwargs.get('detection_method', None)
+        self.detection_method = self.find_detection_tool()
```

Comment:
Do you have a reason to add the ```overwrite_detection_method``` field to the object? It is only used by ```find_detection_tool```, so passing it as a method parameter seems more suitable to me.

    self.detection_method = self.find_detection_tool(kwargs.get('detection_method', None))